File: CS/WpfReportDesigner_CustomizeWizard/MyWizardCustomizationService.cs
using System.Collections.Generic;
using DevExpress.DataAccess.Native.Sql.ConnectionStrategies;
using DevExpress.DataAccess.Wizard.Model;
using DevExpress.DataAccess.Wizard.Presenters;
using DevExpress.Utils.IoC;
using DevExpress.Xpf.DataAccess.DataSourceWizard;
using DevExpress.Xpf.Reports.UserDesigner.ReportWizard;
using DevExpress.Xpf.Reports.UserDesigner.ReportWizard.Pages;
using DevExpress.XtraReports.UI;
using DevExpress.XtraReports.Wizards;
using DevExpress.DataAccess.UI.Wizard;
using DevExpress.XtraReports.Wizards.Presenters;

namespace WpfReportDesigner_CustomizeWizard {
    public class MyWizardCustomizationService : IWizardCustomizationService {
        void IDataSourceWizardCustomizationService.CustomizeDataSourceWizard(DataSourceWizardCustomizationModel customization, ViewModelSourceIntegrityContainer container) {
            if(customization.StartPage == typeof(ChooseDataSourceTypePage<IDataSourceModel>)) {
                customization.Model.DataSourceType = DataSourceType.Xpo;
                customization.StartPage = typeof(ConnectionPropertiesPage<IDataSourceModel>);
            }
            CustomizeProviders(container);
        }

        void IWizardCustomizationService.CustomizeReportWizard(ReportWizardCustomizationModel customization, ViewModelSourceIntegrityContainer container) {
            if(customization.StartPage == typeof(ChooseReportTypePage<XtraReportModel>)) {
                customization.Model.ReportType = ReportType.Standard;
                customization.Model.DataSourceType = DataSourceType.Xpo;
                customization.StartPage = typeof(ConnectionPropertiesPage<XtraReportModel>);
            }
            CustomizeProviders(container);
        }

        bool IDataSourceWizardCustomizationService.TryCreateDataSource(IDataSourceModel model, out object dataSource, out string dataMember) {
            dataSource = null;
            dataMember = null;
            return false;
        }

        bool IWizardCustomizationService.TryCreateReport(XtraReportModel model, out XtraReport report) {
            report = null;
            return false;
        }

        static void CustomizeProviders(IntegrityContainer container) {
            var providers = container.Resolve<List<ProviderLookupItem>>();
            providers.RemoveAll((ProviderLookupItem x) => x.ProviderKey != "MSSqlServer");
        }
    }
}

File: VB/WpfReportDesigner_CustomizeWizard/MyWizardCustomizationService.vb
Imports System.Collections.Generic
Imports DevExpress.DataAccess.Native.Sql.ConnectionStrategies
Imports DevExpress.DataAccess.Wizard.Model
Imports DevExpress.DataAccess.Wizard.Presenters
Imports DevExpress.Utils.IoC
Imports DevExpress.Xpf.DataAccess.DataSourceWizard
Imports DevExpress.Xpf.Reports.UserDesigner.ReportWizard
Imports DevExpress.Xpf.Reports.UserDesigner.ReportWizard.Pages
Imports DevExpress.XtraReports.UI
Imports DevExpress.XtraReports.Wizards

Namespace WpfReportDesigner_CustomizeWizard
    Public Class MyWizardCustomizationService
        Implements IWizardCustomizationService

        Private Sub IDataSourceWizardCustomizationService_CustomizeDataSourceWizard(ByVal customization As DataSourceWizardCustomizationModel, ByVal container As ViewModelSourceIntegrityContainer) Implements IDataSourceWizardCustomizationService.CustomizeDataSourceWizard
            If customization.StartPage Is GetType(ChooseDataSourceTypePage(Of IDataSourceModel)) Then
                customization.Model.DataSourceType = DataSourceType.Xpo
                customization.StartPage = GetType(ConnectionPropertiesPage(Of IDataSourceModel))
            End If
            CustomizeProviders(container)
        End Sub

        Private Sub IWizardCustomizationService_CustomizeReportWizard(ByVal customization As ReportWizardCustomizationModel, ByVal container As ViewModelSourceIntegrityContainer) Implements IWizardCustomizationService.CustomizeReportWizard
            If customization.StartPage Is GetType(ChooseReportTypePage) Then
                customization.Model.ReportType = ReportType.Standard
                customization.Model.DataSourceType = DataSourceType.Xpo
                customization.StartPage = GetType(ConnectionPropertiesPage(Of IDataSourceModel))
            End If
            CustomizeProviders(container)
        End Sub

        Private Function IDataSourceWizardCustomizationService_TryCreateDataSource(ByVal model As IDataSourceModel, ByRef dataSource As Object, ByRef dataMember As String) As Boolean Implements IDataSourceWizardCustomizationService.TryCreateDataSource
            dataSource = Nothing
            dataMember = Nothing
            Return False
        End Function

        Private Function IWizardCustomizationService_TryCreateReport(ByVal model As XtraReportModel, ByRef report As XtraReport) As Boolean Implements IWizardCustomizationService.TryCreateReport
            report = Nothing
            Return False
        End Function

        Private Shared Sub CustomizeProviders(ByVal container As IntegrityContainer)
            Dim providers = container.Resolve(Of List(Of ProviderLookupItem))()
            providers.RemoveAll(Function(x As ProviderLookupItem) x.ProviderKey <> "MSSqlServer")
        End Sub
    End Class
End Namespace
